src/python/pants/engine/internals/graph_test.py

```diff
@@ -844,6 +847,72 @@ class IrrelevantSources(Sources):
 # -----------------------------------------------------------------------------------------------
 
 
+def test_parse_dependencies_field() -> None:
+    result = parse_dependencies_field(
+        [
+            ":relative",
+            "//:top_level",
+            "demo:tgt",
+            "demo",
+            "./relative.txt",
+            "./child/f.txt",
+            "demo/f.txt",
+            "//top_level.txt",
+            "top_level2.txt",
```

Comment:
I don't like this and I think prefer requiring `//`, but I also don't think it's worth the special casing that we would now need to explain in docs and making our heuristic that much more complex.